### PR TITLE
treewide: replace search.cpan.org with metacpan.org

### PR DESCRIPTION
**Pull request by ryantm**
https://www.perl.com/article/saying-goodbye-to-search-cpan-org/

### CODE REVIEW

**Comment by ryantm** [comment]
I didn't update `pkgs/development/perl-modules/expression-generator/` which was edited or created? by @7c6f434c

**Comment by 7c6f434c** [comment]
`cpan2nix` is currently used for most of the Perl packaging in Nixpkgs,
right? I forgot to drop these scripts as I stopped trying to package
Perl packages.

**Comment by xeji** [comment]
cc @volth

**Comment by xeji** [comment]
I think it makes more sense to merge this to staging and resolve conflicts now rather than having them pop up later in the staging->master merge.

**Comment by volth** [comment]
Another option might be to remove all ```search.cpan.org```/```metacpan.org``` urls from ```meta.homepage``` and set a generic default homepage inside ```bulidPerlPackage```

(something like
 ```
meta.homepage = "http://metacpan.org/release/${lib.removeSuffix "-${version}" name}";
```
)

**Comment by ryantm** [comment]
I prefer @volth's solution, but let's use https.

**Comment by ryantm** [comment]
Can you check the new commit @volth?

File: pkgs/development/perl-modules/generic/default.nix [hunk 1]
```diff
@@ -20,6 +20,8 @@ perl.stdenv.mkDerivation (
     # authors to skip certain tests (or include certain tests) when
     # the results are not being monitored by a human being."
     AUTOMATED_TESTING = true;
+
```
Comment (volth) [inline pkgs/development/perl-modules/generic/default.nix]: I have not tried to run it, but at the first glance it looks like that any ```meta``` (say ```meta.license``` from ```attrs``` below would destroy this ```meta.homepage```

Comment (ryantm) [reply pkgs/development/perl-modules/generic/default.nix]: Oh, yep, it does :(

Comment (ryantm) [reply pkgs/development/perl-modules/generic/default.nix]: I happened to try testing a package that only had that one meta attribute.

**Comment by ryantm** [comment]
@volth Okay, what I have now works, but it seems kind of ugly to me. What do you think?

**Comment by volth** [comment]
maybe ```lib.recursiveUpdate```? not sure if it nicer.

**Comment by ryantm** [comment]
@volth Yes, that seems nicer!

**Comment by xeji** [comment]
now should this go to master or staging?

**Comment by volth** [comment]
@xeji I guess staging (or wait until staging merged to master), to avoid big merge conflict
And let's try to remove 2 brackets first https://github.com/ryantm/nixpkgs/pull/1 :)

**Comment by ryantm** [comment]
@volth I incorporated your patch and switched the PR over to staging

**Pull request merged**